### PR TITLE
Refactor/energy system (serialize-instruction-data-frontend.md)

### PR DESCRIPTION
**Pull request by EmekaManuel**
### Problem
- The code for fetching and managing player data had inconsistent use of .then() for handling promises.
- Error handling was not robust, leading to potential issues if data fetching or decoding fails.
- Account change subscription lacked proper cleanup, which could cause memory leaks.

### Summary of Changes
- Refactored code to use async/await for consistency and readability.
- Added try/catch blocks for improved error handling when fetching and decoding player data.
- Created a separate async function for fetching player data to enhance code organization.
- Implemented error handling in the account change callback to handle decoding errors.
- Added cleanup logic for the account change subscription to prevent memory leaks.

### CODE REVIEW

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md [hunk 1]
```diff
@@ -255,11 +255,18 @@ const instruction = new web3.TransactionInstruction({
...
-web3.sendAndConfirmTransaction(connection, transaction, [player]).then(txid => {
+try {
+  const transactionId = await web3.sendAndConfirmTransaction(
+    connection,
+    transaction,
+    [player],
+  );
   console.log(
-    `Transaction submitted: https://explorer.solana.com/tx/${txid}?cluster=devnet`,
+    `Transaction submitted: https://explorer.solana.com/tx/${transactionId}?cluster=devnet`,
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/serialize-instruction-data-frontend.md:265]: Check out https://github.com/solana-developers/helpers?tab=readme-ov-file#get-a-solana-explorer-link-for-a-transaction-address-or-block

Comment (mikemaccana) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md:265]: We should also move to just importing what we need from web3 (it's more efficient due to tree shaking and stops us needing to type `web3.` all the time).

**Comment by mikemaccana** [comment]
Thanks for these improvements @EmekaManuel ! Just a note that there's already an older PR for this content, see https://github.com/solana-foundation/developer-content/pull/348

File: content/courses/native-onchain-development/serialize-instruction-data-frontend.md (hunk 1 above)
Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md:265]: > We should also move to just importing what we need from web3 (it's more efficient due to tree shaking and stops us needing to type `web3.` all the time).

yes i have made this change in all the PR's - #346, #347 , #348

Comment (EmekaManuel) [reply content/courses/native-onchain-development/serialize-instruction-data-frontend.md:265]: > Check out https://github.com/solana-developers/helpers?tab=readme-ov-file#get-a-solana-explorer-link-for-a-transaction-address-or-block

implemented ✅

**Comment by EmekaManuel** [comment]
> Thanks for these improvements @EmekaManuel ! I note you already have a PR for this content, see #348 - can you move these changes into that?

Done ✅

**Review by mikemaccana — APPROVED** [review]

**Comment by mikemaccana** [comment]
Closing duplicate in favour of #348

**Pull request merged**